rockcraft/oci.py

```diff
@@ -215,34 +216,56 @@ def to_oci_archive(self, tag: str, filename: str) -> None:
         src_path = self.path / f"{name}:{tag}"
         _copy_image(f"oci:{str(src_path)}", f"oci-archive:{filename}:{tag}")
 
-    def set_entrypoint(self, entrypoint: List[str]) -> None:
-        """Set the OCI image entrypoint.
-
-        :param entrypoint: The default list of arguments to use as the command to
-            execute when the container starts.
-        """
+    def set_entrypoint(self) -> None:
+        """Set the OCI image entrypoint. It is always Pebble and CMD is null."""
         emit.progress("Configuring entrypoint...")
         image_path = self.path / self.image_name
+        entrypoint = [f"/{Pebble.PEBBLE_BINARY_PATH}", "enter"]
         params = ["--clear=config.entrypoint"]
         for entry in entrypoint:
             params.extend(["--config.entrypoint", entry])
         _config_image(image_path, params)
+        # Clear the CMD
+        _config_image(image_path, ["--clear=config.cmd"])
         emit.progress(f"Entrypoint set to {entrypoint}", permanent=True)
 
-    def set_cmd(self, cmd: List[str]) -> None:
-        """Set the OCI image default command parameters.
-
-        :param cmd: The default arguments to the entrypoint of the container. If
-            entrypoint is not defined, the first entry of cmd is the executable to
-            run when the container starts.
+    def set_pebble_services(
+        self,
+        services: Dict[str, Any],
+        name: str,
+        tag: str,
+        summary: str,
+        description: str,
+        base_layer_dir: Path,
+    ) -> None:
+        """Write the provided services into a Pebble layer in the filesystem.
+
+        :param services: The Pebble services to be written into the Pebble
+        :param name: The name of the ROCK
+        :param tag: The ROCK's image tag
+        :param summary: The summary for the Pebble layer
+        :param description: The description for the Pebble layer
+        :param base_layer_dir: Path to the base layer's root filesystem
         """
-        emit.progress("Configuring cmd...")
-        image_path = self.path / self.image_name
-        params = ["--clear=config.cmd"]
-        for entry in cmd:
-            params.extend(["--config.cmd", entry])
-        _config_image(image_path, params)
-        emit.progress(f"Cmd set to {cmd}", permanent=True)
+        # pylint: disable=too-many-arguments
+        service_names = list(services.keys())
+        emit.progress(f"Configuring Pebble services {', '.join(service_names)}")
+        pebble_layer_content = {
+            "summary": summary,
+            "description": description,
+            "services": services,
+        }
+
+        pebble = Pebble()
+        with tempfile.TemporaryDirectory() as tmpfs:
+            tmpfs_path = Path(tmpfs)
+            pebble.define_pebble_layer(
+                tmpfs_path, base_layer_dir, pebble_layer_content, name
+            )
+
+        emit.progress("Writing new Pebble layer file")
+        self.add_layer(tag, tmpfs_path)
```

Comment:
something is weird here; the `TemporaryDirectory()` context manager is supposed to remove the directory on cleanup, so how are we still able to access it here??
(imo these two lines should be inside the context manager, and the `shutil.rmtree()` call can be removed.

Reply:
hum...true.

I guess the ongoing spread tests will tell us whether that is the case or not. 

I'm applying those changes anyway